src/app/charts/line-graph.component.ts

```diff
@@ -196,11 +219,12 @@ export class LineGraphComponent {
             if (this.min && this.minVal) {
                 // Draw min threshold line
                 let minData = [{'date': x1, 'value': this.minVal}, {'date': x2, 'value': this.minVal}];
-                this.drawLine(minData, 'minline');
+                this.drawLine(minData, 'min-threshold');
 
                 // Prepare data for bar graph
+                let yDataCopy = _.cloneDeep(this.yData);
                 let minBars = _.map(this.extractedData, datum => {
-                    let val = this.minVal > datum['value'] ? y * 2 : 0;
+                    let val = this.minVal >  yDataCopy.shift() ? y * 2 : 0;
```

Comment:
I continue to find this shift-from-a-copy construction odd, though it does work fine.

One possible alternative, obviously, would just be using `datum.values.avg`.  If it's desirable to use `yData` (which I could see, to avoid adding two more references to the specific structure of the response data), another option would be to use the index provided by `_.map`, i.e.
```
let minBars = _.map(this.extractedData, (datum, index) => {
    let val = this.minVal >  this.yData[index] ? y * 2 : 0;
    ...
```